app/models/pageflow/entry_template.rb

```diff
@@ -4,17 +4,22 @@ class EntryTemplate < ApplicationRecord
     include SerializedConfiguration
     serialize :default_share_providers, JSON
     belongs_to :account
+    delegate :enabled_feature_names, to: :account
     has_many :widgets, as: :subject, dependent: :destroy
 
     validates :account, presence: true
-    validates :entry_type, presence: true
-    validates :entry_type,
+    validates :entry_type_name, presence: true
+    validates :entry_type_name,
               uniqueness: {
                 scope: :account
               }
 
-    def translated_entry_type
-      I18n.t("activerecord.values.pageflow/entry.type_names.#{entry_type}")
+    def entry_type
+      Pageflow.config.entry_types.find_by_name!(entry_type_name)
+    end
+
+    def translated_entry_type_name
+      I18n.t("activerecord.values.pageflow/entry.type_names.#{entry_type_name}")
     end
 
     def resolve_widgets(options = {})
```

Comment:
**issue**: Theme validation does not consider entry type

I just saw this now and cannot comment on the correct line since it is not part of the change, but `available_themes` a few lines below still uses `config_for(account)`, which allows using all themes, paged and scrolled, that are available for the account, even though the select box only offers a subset. In Dedicated we now already have examples of themes that are registered for specific entry types.

Reply:
For `resolve_widgets`, this should also be fixed, right?

Reply:
> For `resolve_widgets`, this should also be fixed, right?

PS: Only usage of `resolve_widgets` on entry templates is for the editor

Reply:
Hm. Not even sure this is used. But yes, would make sense to fix it there as well.

Reply:
There are definitely widgets with widget type `Pageflow::EntryTemplate` in the database. My guess is that we list widgets that are available for batch update using `index` action in `WidgetsController`

Reply:
One more thing: We actually got a failing spec after switching `available_themes` to entry template config, because entry templates factory uses `'default'` theme. I would fix this by adding `config.themes.register(:default)` to `spec/support/pageflow/dummy/config/pageflow.rb`. What do you think?

Reply:
That's a bit too global for my taste. I faced a similar problem when introducing entry type specific themes and therefore extended the `TestEntryType` helper to [register a `default` theme](https://github.com/codevise/pageflow/blob/76d95420e2608c9488ced3854adf70fa58e0e2de/spec/support/helpers/test_entry_type.rb#L7). Can that maybe also be used in this situation?

Reply:
Ah yes, that sounds like a better idea. I'll try that then